generators/excel/main.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+import os
+import csv
+import openpyxl
+import sys
+from datetime import datetime
+
+OPENPYXL_MAJOR_VERSION = int(openpyxl.__version__[0])
+DATETIME_FORMAT = '%Y-%m-%dT%H:%M:%S%Z'
+# Minimum possible header.
+HEADER = (
+    'domain',
+    'entity',
+    'attribute',
+    'value',
+)
+
+def generate(file_name, domain):
+   # Creating a CSV writer that feeds to stdout.
+    w = csv.writer(sys.stdout)
+
+    # Write the header to start.
+    w.writerow(HEADER)
+
+    wb = openpyxl.load_workbook(file_name, use_iterators=True)
+    sheets = wb.get_sheet_names()
+    name = os.path.splitext(os.path.basename(file_name))[0]
+
+    for i, sheet_name in enumerate(sheets):
+        sheet = wb.get_sheet_by_name(sheet_name)
+        sheet_domain = domain + '.' + sheet_name.lower().replace(' ', '_')
+        column_names =  _column_names(wb, sheet_name)
+        rows = sheet.iter_rows()
+
+        # Skip header row because we already collected the column names.
+        next(rows)
+        rows = list(rows)
+        for j, row in enumerate(rows):
+            row = rows[j]
+
+            for k, cell in enumerate(row):
+                if cell.value is not None:
+                    w.writerow([
+                        sheet_domain,
+                        j,
+                        column_names[k],
+                        cell.value,
+                    ])
+
+def _column_names(workbook, sheet_name):
+    sheet = workbook.get_sheet_by_name(sheet_name)
+    first_row = next(sheet.iter_rows())
+
+    # Renamed attribute in version 2+
+    if OPENPYXL_MAJOR_VERSION > 1:
+        return [c.value for c in first_row]
+
+    return [c.internal_value for c in first_row]
+
+def main():
+    usage = """Origins Excel Generator.
+    Usage:
+        excel <file_name> [--domain=<domain>]
```

Comment:
Add the `--identifier` option like in the [delimited generator](https://github.com/chop-dbhi/origins-generators/blob/master/generators/delimited/main.py#L68) for specifying which columns define the entity names. However, see it should be implemented the way described here: https://github.com/chop-dbhi/origins-generators/issues/10